src/logic/calculate.js

```diff
@@ -0,0 +1,30 @@
+import operate from './operate';
+
+const calculate = (data, buttonName) => {
+  let { total, next, operation } = data;
+
+  switch (buttonName) {
+    case '+/-':
+      total *= -1;
+      next *= -1;
+      break;
+    case '=':
+    case '%':
+    case '+':
+    case '-':
+    case '÷':
+    case 'x':
+      total = operate(total, next, operation);
+      break;
```

Comment:
[Optional] It would have been nice if you added logic for the operations as well.